community/security/src/main/java/org/neo4j/server/security/auth/UserSerialization.java

```diff
@@ -115,11 +129,18 @@ private Credential deserializeCredentials( String part, int lineNumber ) throws
 
     private String join( String separator, String... segments )
     {
+        return join(separator, Arrays.asList(segments).iterator() );
+    }
+
+    private String join( String separator, Iterator<String> segments )
+    {
```

Comment:
I think maybe we should guarantee the sort order of flags when serializing (like we do with roles)